Cython/Compiler/Code.py

```diff
@@ -2332,8 +2335,17 @@ def put_pymethoddef_wrapper(self, entry):
             if method_noargs in method_flags:
                 # Special NOARGS methods really take no arguments besides 'self', but PyCFunction expects one.
                 func_cname = Naming.method_wrapper_prefix + func_cname
-                self.putln("static PyObject *%s(PyObject *self, CYTHON_UNUSED PyObject *arg) {return %s(self);}" % (
-                    func_cname, entry.func_cname))
+                body = "%s(self)" % entry.func_cname
```

Comment:
Let's not call this `body`. It confused me since I was excepting a function body, until I realised it was only a function call, and then it started changing meaning along the way.

Also, there is no reason why all of this has to go into a single C code line, right? Chaining an `if` in C with unrelated code seems very brittle. Just generate the code sequentially, as usual, and all problems should go away.